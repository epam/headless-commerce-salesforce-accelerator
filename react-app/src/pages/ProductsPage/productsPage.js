/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useEffect, useMemo } from "react";
import { useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { Grid } from "@material-ui/core";

import productsActions from "store/actions/products/actions";

import filterRefinementsActions from "store/actions/filters/actions";
import {
  selectProductsIds,
  selectProductsCategory,
  selectProductsResultCount,
  selectIsProductsLoading,
} from "store/selectors/products/selectors";
import {
  selectRefinements,
  selectSelectedRefinements,
} from "store/selectors/filters/selectors";

import FiltersPanel from "components/FiltersPanel";
import ProductGrid from "components/ProductsGrid";
import Pagination from "components/Pagination";
import ScrollToTopController from "components/core/ScrollToTopController";
import StyledProductsPage from "./productsPageStyled";
import SearchResultsLabel from "./searchResultsLabel";
import StyledSearchResults from "./searchResultsStyled";

const ProductsPage = () => {
  const dispatch = useDispatch();
  const { search } = useLocation();

  const productsIds = useSelector(selectProductsIds);
  const category = useSelector(selectProductsCategory);
  const resultsCount = useSelector(selectProductsResultCount);
  const refinements = useSelector(selectRefinements);
  const selectedRefinements = useSelector(selectSelectedRefinements);
  const isProductsLoading = useSelector(selectIsProductsLoading);

  useEffect(() => {
    dispatch(productsActions.getProducts(search));
    dispatch(filterRefinementsActions.getFilterRefinementsStart(search));
  }, [dispatch, search]);

  const productsGrid = useMemo(
    () => (
      <Grid item xs={9}>
        <Grid container justify="center">
          <ProductGrid
            productsIds={productsIds}
            isProductsLoading={isProductsLoading}
          />
        </Grid>
      </Grid>
    ),
    [isProductsLoading, productsIds]
  );

  const filtersPanel = useMemo(
    () => (
      <Grid item xs={3}>
        <FiltersPanel
          refinements={refinements}
          selectedRefinements={selectedRefinements}
          searchUrl={search}
        />
      </Grid>
    ),
    [refinements, search, selectedRefinements]
  );

  return (
    <StyledProductsPage>
      <ScrollToTopController />
      <div>
        <SearchResultsLabel query={category} amount={resultsCount} />
      </div>

      <StyledSearchResults>
        <Grid container>
          {filtersPanel}
          {productsGrid}
        </Grid>
      </StyledSearchResults>
      <Pagination />
    </StyledProductsPage>
  );
};

export default ProductsPage;
