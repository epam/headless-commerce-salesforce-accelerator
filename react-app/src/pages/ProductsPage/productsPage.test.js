/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import PropTypes from "prop-types";

import SearchResultsLabel from "./searchResultsLabel";
import ProductsPage from "./productsPage";

import {
  StyledSearchResultsLabel,
  StyledSearchQuery,
  StyledSearchAmount,
} from "./searchResultsLabelStyled";
import StyledSearchResults from "./searchResultsStyled";
import getMainTheme from "../../theme";

const ReduxProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

ReduxProvider.propTypes = {
  children: PropTypes.element,
  reduxStore: PropTypes.shape({}),
};

const theme = getMainTheme();

describe("SearchResultsLabel", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<SearchResultsLabel query="Bags" amount={158} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe("ProductsPage", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <ProductsPage />
      </ReduxProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledSearchResultsLabel", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<StyledSearchResultsLabel theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledSearchQuery", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<StyledSearchQuery theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledSearchAmount", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<StyledSearchAmount theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledSearchResults", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<StyledSearchResults theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });
});
