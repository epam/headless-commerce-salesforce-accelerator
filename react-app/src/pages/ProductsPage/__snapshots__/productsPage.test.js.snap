// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPage should match snapshot 1`] = `
<Provider>
  <ProductsPage />
</Provider>
`;

exports[`SearchResultsLabel should match snapshot 1`] = `
<styled.div>
  <styled.span>
    Bags
  </styled.span>
  <styled.span>
    158
  </styled.span>
</styled.div>
`;

exports[`StyledSearchAmount should match snapshot 1`] = `
<span
  className="sc-AxiKw cHYbPL"
/>
`;

exports[`StyledSearchQuery should match snapshot 1`] = `
<span
  className="sc-AxirZ ekKuNX"
/>
`;

exports[`StyledSearchResults should match snapshot 1`] = `
<div
  className="sc-fzoyAV jeiXBX"
/>
`;

exports[`StyledSearchResultsLabel should match snapshot 1`] = `
<div
  className="sc-AxjAm CybCx"
/>
`;
